File: notification-service/src/main/java/com/bank/notificationservice/service/metric/MetricRegistry.java
package com.bank.notificationservice.service.metric;

import com.bank.notificationservice.dto.kafka.MetricDto;
import com.bank.notificationservice.service.kafka.sender.KafkaMetricSender;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class MetricRegistry {
    private final KafkaMetricSender kafkaMetricSender;

    public void register(MetricDto dto) {
        kafkaMetricSender.send(dto);
    }
}

File: core-query/src/main/kotlin/ru/patterns/corequery/utils/metric/MetricRegistry.kt
package ru.patterns.corequery.utils.metric

import org.springframework.stereotype.Component
import ru.patterns.corequery.utils.metric.serialization.MetricDto

@Component
class MetricRegistry(
    private val kafkaMetricSender: KafkaMetricSender
) {
    fun register(metricDto: MetricDto) {
        kafkaMetricSender.sendObjectToKafkaAsync(metricDto)
    }
}